profiles/lib/relations.lua

```diff
@@ -258,4 +258,18 @@ function Relations.process_way_refs(way, relations, result)
   end
 end
 
+function Relations.filter_relations(relations, way, key, value, ret, forward)
+  -- if any of way's relation have key=value, return tag ret; else return nil
+  -- todo: check backward and forward
```

Comment:
is forward/backward actually handled by this function?

Reply:
no, my lua knowledge is not that good. (plus I don't know of any such route in my local area).

could you add that check?